icicle/src/runtime.cpp

```diff
@@ -274,9 +274,28 @@ extern "C" eIcicleError icicle_load_backend(const char* path, bool is_recursive)
   };
 
   auto load_library = [](const char* filePath) {
-    ICICLE_LOG_DEBUG << "Attempting load: " << filePath;
-    void* handle = dlopen(filePath, RTLD_LAZY | RTLD_GLOBAL);
-    if (!handle) { ICICLE_LOG_ERROR << "Failed to load " << filePath << ": " << dlerror(); }
+    // Convert the file path to a std::string for easier manipulation
+    std::string path(filePath);
+
+    // Extract the library name from the full path
+    std::string fileName = path.substr(path.find_last_of("/\\") + 1);
+
+    // Check if the library name contains "icicle" and if the path contains "/backend/"
+    if (fileName.find("icicle") == std::string::npos || path.find("/backend/") == std::string::npos) {
```

Comment:
Does that mean that library should always be inside "/backend/" folder?

Reply:
the backend libs yes.